Gemfile

```diff
@@ -23,6 +23,12 @@ gem 'puppet', *location_for(ENV['PUPPET_VERSION']) if ENV['PUPPET_VERSION']
 
 group :test do
   gem 'rspec'
+  # dependency for Puppet on Ruby 3.4
```

Comment:
We only mention these dependencies in test but `puppet` itself is listed on the top level. That feels a bit inconsistent.

Reply:
moved it out of the test group